python/GafferImageUI/ImageInspector.py

```diff
@@ -0,0 +1,460 @@
+
+	ImageInspector.Settings,
+
+	plugs = {
+
+		"view" : [
+
+			"plugValueWidget:type", "GafferImageUI.ViewPlugValueWidget",
+			"description", "The view to inspect",
+
+		],
+
+	}
+
+)
+
+##########################################################################
+# _ImagePath
+##########################################################################
+
+class _ImagePath( Gaffer.Path ) :
+
+	def __init__( self, scriptNode, image, context, path, root = "/", filter = None ) :
+
+		Gaffer.Path.__init__( self, path, root, filter )
+
+		self.__scriptNode = scriptNode
+		self._context = context
+		self._image = image
+		self.__imagePlugDirtiedConnection = self._image.node().plugDirtiedSignal().connect(
+			Gaffer.WeakMethod( self.__imagePlugDirtied, fallbackResult = None ), scoped = True
+		)
+
+	def copy( self ) :
+
+		return self.__class__( self.__scriptNode, self._image, self._context, self[:], self.root(), self.getFilter() )
+
+	def cancellationSubject( self ) :
+
+		## \todo We really just want to return `self._image` here, but we
+		# can't because BackgroundTask can't find the right ScriptNode from it.
+		return self.__scriptNode["fileName"]
+
+	def _children( self, canceller ) :
+
+		return [
+			self.__class__( self.__scriptNode, self._image, self._context, self[:] + [ name ], self.root(), self.getFilter() )
+			for name in self._childNames( canceller )
+		]
+
+	def _childNames( self, canceller ) :
+
+		raise NotImplementedError
+
+	def __imagePlugDirtied( self, plug ) :
+
+		if plug.parent() == self._image and self._imagePlugAffectsPath( plug ) :
+			self._emitPathChanged()
+
+##########################################################################
+# _FormatPath
+##########################################################################
+
+class _FormatPath( _ImagePath ) :
+
+	def __init__( self, scriptNode, image, context, path, root = "/", filter = None ) :
+
+		_ImagePath.__init__( self, scriptNode, image, context, path, root, filter )
+
+	def propertyNames( self, canceller = None ) :
+
+		return Gaffer.Path.propertyNames( self ) + [ "format:value" ]
+
+	def property( self, name, canceller = None ) :
+
+		if name != "format:value" :
+			return Gaffer.Path.property( self, name, canceller )
+
+		if len( self ) != 1 :
+			return None
+
+		with Gaffer.Context( self._context, canceller ) :
+
+			if not self._image.channelNames() :
+				# Heuristic for detecting lack of input image.
+				return "---"
+
+			if self[0] == "Name" :
+				return GafferImage.Format.name( self._image.format() ) or "---"
+			elif self[0] == "Resolution" :
+				size = self._image.format().getDisplayWindow().size()
+				return f"{size.x} x {size.y}"
+			elif self[0] == "Pixel Aspect" :
+				return self._image.format().getPixelAspect()
+			elif self[0] == "Origin" :
+				origin = self._image.format().getDisplayWindow().min()
+				return f"{origin.x}, {origin.y}"
+			elif self[0] == "Data Window Origin" :
+				origin = self._image.dataWindow().min()
```

Comment:
I think having the data window max would be useful and canonical.

Reply:
Added in https://github.com/GafferHQ/gaffer/pull/5598/commits/22ec8a5c3f62905182341005837d316c765e5162. I also renamed the "Origin" fields to "Min" to better associate them with "Max".